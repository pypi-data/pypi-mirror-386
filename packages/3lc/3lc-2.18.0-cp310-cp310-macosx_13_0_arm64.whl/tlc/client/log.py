# Pyarmor 9.1.9 (pro), 005329, 3lc, 2025-10-22T14:50:27.062415
from tlcsaas.pyarmor_runtime_005329 import __pyarmor__
__pyarmor__(__name__, __file__, b'PY005329\x00\x03\n\x00o\r\r\n\x80\x00\x01\x00\x08\x00\x00\x00\x04\x00\x00\x00@\x00\x00\x00\xa6\x07\x00\x00\x12\t\x06\x00\x92\xe2\x9aWR\x1dY\xe7\xd1\r\\\xfc\rP\xb0\x01\x00\x00\x00\x00\x00\x00\x00\x00\xbau\x85\xac\x16\xaa."F\t\xe7\xcf\x11\\\x08(\xef,y\x922\xaa\xa3\x94\xfb\xad,\x03\x0e\xec\x14\x85\x07\r\xe3\x8eeGX\xf7\xfdEo\xbe\x98p\xc3L\xa36\xa0\xbb\xe4\xaa\xcd\x08\xf4\x1fn\x89\xae?G\xc2w\x8aF\xf6\xe1\xc2[\xea\x14R\xf3\xa8s\xb4P`\x9b\x88m\xa6\xcas-\xe5\x80Q\xf0\x9e\xb81\xc8\xb8\xeb\x127\x8d\x06\xc7\xe3Z\x18\x922\x00\x94\x8e\x86$\xe7\xdc\xdboC\x1f\xea0\x82\xca\x9f\xb9\x9cI\x92{.W0\xf6u\xb3\xe6\xcdGp\xd2\xe3K\xb7X\xe1eA\x9a{\xccm2\xc3\x15\xc8\xa1x\xe1\x9b\x0b\x17\xe6\x11\x92\x1d\xc4LO"x\x8a\xad\x820p\x92\x11p\xcf\xde\xd5\xf2\xe9v\x1f\xd93\xdd\xe4\x17\x83Q\x9e\xe2\x87\xdb:4a\xee\xaa\x14X\xb3\x98Z\xa5\x85\xcc\x98\x17\xab\xc0\xd4-bU&\xc7\xe2\\\r\x96\xd2\xb6$\xce\xd7f\xaf\x81R\xb8K\x98}z\xa6\x90\xdc\x9ca=\x02\xfd\x1c\x1e\xc3\xff\xb4g\xfdXh\xb7\x93\x0cMd\x14\xbc\xc5\x00\xac\x07&!D\x8d\x94\xaf\xa0\x0c\xe7\xc0\xae\xd4\xbeB\xb6\xe4\x11\\\x03\x8di\x15\x8f\x1c\xbd\xb9\xb5\xc4\xec\x16\x7f\xcflQ\x0e\x10\x19e3MU3\xc5X\xa3\xf2\xca\xb3\xf9#^\x80X;\xb6$\xa3\x8c\xbc\xf0\xc7\'\xaabCG\xb0<\xa4a\xf4\xa8\xc5\x18\x08\xaf\xba2q\xa4\x05\xc2\xaf\xdd\xcb\xb2\x82\x8ax\x01\xf9\x00\xd4;\xb3l\\\xc5a\xd4\x9d\x14\xcc\xdb/OO\xd8W\xd8\xc4h\xd5\x83")D\xe3?\xe2\xbfd\xbcHxo\xe4\x9b\r\x06\x04\x88E\x9a\x0b\xda\x9c\x1a\x84XJs\xd9\xa8eG4\x15\xb2\x02\xd2\x0e\x8d\xd3\xa1b\x1a;\x92\x8bi\x9b\xc8\xe8&\xd7\x90]#:c\xc4\xa4\xc2\x12I\xa9\xc5D\xa2\x19\xc6\x11\x84hi]E\xd9\xb3\xe6\xf5=\xb0\xa1<NJ\xdf\xe2\xebf\xcf\xfc\x16Tc\x94\xb3\xcb_LGRM\xaey\x19\xa1n\xf4\xbb\x1a\x9c\xc1R\x0e\x94\xe3\xb9T[M\x04\x1a\xcb\xd4\xb9\xd4c\xe5\x17\xd8m\x19\xd7\xde\xa8\x989u\xb4\n\x1d\xc8\x00\x8a\xa7\x8f\xdc\x1b\xbaP\xdd\\Z\xd9\x82\x18\xe9\xa8\x02Hl\xf2\xb52D\xa1\xb8I\x86/a\xe1\xc7\xdei\x8e\xa1\xd1\xb8%\xf6}\xcfK\x99k"\x0cC\xedl#\xc4I\xba15e\x10_\x90\xa1\xa5\xb7\xccI\x11\xc4\xd7_vW\x00\xad\x18\xbdaI\x0c\xe7\xce\xb2N\xaf\x98\x1e\xed\xa4\n\xaf\xf40h\x14\x1e+\x00(C\xaf6\xa7q\xc6\xe05\xeeIE\x92\xf2\x13\xfb\x05\xc2:o\xaf\x9d\xff\x83\xc3\x11\\_U\xbc\x00\xd2O\xab\xe3\x8e\xa0g\xd7\xe2\x00\x02\x00\x98\xae\x90\xa6I\xfbf(\xb5\x906\t\xecp\xbf\xa7eO\xb2\xcb@\xeb\x1c\xa8\x05J\x97\xd3\xe2\x14V\xca~\x8c/9\xcc\xb3k\x0e\xf5+\x8f\x02(q\x00\xfc\xcb\x97\x85\x13h\x90F\x9f\xe3k\x00\xd0: \xd9\x97;v\xd0\x93,\xf5\xe4jSJ]2\x17:\xe5\xec\xa5\x87\x82cZMG\x12B\xa4:\xb1\xcb\xd0\x98!g\xeax\x1e\xb3\xbe\x00\xdb\t\x9d\xfb\xaf\xed\x92\x80\xdb\xdb\xc6\xb5\xb4\xb4\xd9\xff\x82\xde\xf6\xe8]\xf5\xb7)\xef\x0b\x9d\x95k\xdac\x1d\x1a\xb5,4\xaeo\xae\x06\xa6\x1cdj\xd7\x02\x8f0\xf3\xd9i(\x9b\xebOz\xf1\x84\xa0B^\xb9\xf6 \xe2Q4~t\n\x15%]\x04\xce\xc0\xbb$_\xcfI{H\xb6\xb3\xec;\xa8\xc7>\x1d!,\xd6\x8a\xd3\x8d\xbb[z;1\xaf\xaf cl\x8d\r6\x97%\xeb\xf4L\x12\x92\x03?\xfa\xc1\xf2D\x92\x88?u\xa4WA\x8a\x88\xc9T1Z\xf0\x05[YX\xf9\xd6\xef\x8b\xad?A\xe1\xacf\x0cp~t\xf9\xe4\xe1\x14\xa1^N%\xf5Nm\xfd\xece_\xfd"\xce\x93T\r\nY\xe4\xabo:\xa3\x94,U\xc3\x85\x1da\x0c\x1b=\x91\x90[\x89ogk\x15\xc3\xfc \xce\xa3/\xe6\x99\xf7\xe4\xbbq\x94\xd9\x07y\xf2\x1d "\x84o\xd2\xcc\x1c\x87\xc3~hPTJ\xb9\x03"Z\xca\xff;\x94\xbd_\\\xfc3b\x0c\xd0H\x00]\xb79bJ\xf4\xd3\xf3&\xf0\x12\x0f\xa7!\xa2\x1e\x14\x98\x92\x9cEK\xd5\xfb\x8b#\x01\x8fgh\x184\x15\xccec\x08:D\xd9\x03\xb0\x10\x0e\xbd{,\xdeT]\x15sC\x8f\x04\xb2\x02\x00\xe7KT\x0b\xbaz\xea\xae\xbfNAQj6X3\x9cX,\xee\xcdK\xa9{\x14V,\x01\xcd-}\x91j\x16T\xa1K\xae{\xea\xfd\xf3\xed\x0b\x18\xd1\xe5jp\x89O\x9a\xd5?CUw\xc0Z\xf8\x18h\x02\x84<\xaf\x16\x85\x013\xecR\xd8\x8b|\x1c\xda\x7fLC\x1a9e!O\x9c\xc95Y\xbcR\x8a\xf3g3\xae)a=\x8f\x19\x04i\x93)\xe3\xcb$c\x1e\x8arB\xdc\xa3h@\x1f|\x01yS\x01\xbf\xc6{z%\xae\xe8\'\x9e\xa7\xd4\xaf\xf9]\xbc\xe32\x9f\x89\xcbz\xb3d:o\xe8+x$\xbf1,\xce[\xd1\xb4AX\xe2x\x8b\xee\x11\x02\xbe\xdc\x9f\xbc\xadJSkg\xa8\x8e\xd50\x02f6a\x02Y\xf9\xac\xe63Q\xea\x7f\xa4QU\xbf\xa6\x0bz}\xd6\xf2\x0f\x1d\xca\xa5\xce\xc4\xae[!\x01\x15KCk\'.\x82O^o\xc7\xe8\xaem\xde\xab;(:\x9c\xf2\x83\x91&%\xc4=X\xd9\x17\xa1\xc8G\xcdT\x1a\x06\xf1\\\x05\xa0\xd4R\x99/\x912wuh\xca\xc2O\'\xd2\x04%\xd1\xd5\xb6@\x18(][I\x8b3\x1c\x86\xe2@\xe7\x189i\x14\xc4\xf8\xb2c\x9c\x03^\x11\xbe\xf9\xa1\xa4V|\x87\xc3\x14\x15\xfa\xe5\xf5mf\xeb\x17|\x89\xea\x1f:\x9d\xbe\x96\xb5l\xe7\xf5\xba\x9f#-\xcb.\xff\xd49\xf6\xb5\x1a\x10>H\x91\x16\x14\x83/\x98\xfa\xc7\xd5\xbe`\xc3\xa8\xb4!p\x1b\xabEuR\x92\xc5#YFJ\n\x89\xbd8Bj\xe4\x866\xda\xaaY\x9a,E\xba\xfa\xab/\x1f\xa0\xc2\x86Th+\xcb>\xae4\xd7\x879\x83\x1e\x0e\x84\xd9gw\xc3\xbdI\xf1J\xd3\xfcO\xe9\xd5\xde\xf5\n\x7fs\xf8Ar\x8aP.\x8e\x86\xbf\xfe\x9e\x80\xe6;\xc8\xech\x89Y"\xac8c\xa2\x82\x17^`w\xcb\xd3\xee\xf0m[0\xdb{\xe8\x95\x99)w\xe9Y\xb0\xb2\x198\xc2\x05\xb7\xd6\xe7\x90\x18\x12\x8b\xac0\x82HI\xac\x8b\x87\xdd\x93Q7\x83\n\xce)}\xa2\x056\xbc\xef\xf7Z\xa4H\xc3A3\xe1\x8e\xaa\xd3\xbeO|\x8d\xb0"\'y\xd6\xc1\x8d;\xf9\x06\xad\xe1q\x05\x07\x8c\xfc<N\xa3\xdb\xa1}v\xb2\x9a\xbc"\xc9\x1c\xfc\xc9Y7\x11\xf9\x9f\xdc\x85\xd6\xed\xf4c6\xd2*M\xb3b\xa82\xf4\x07ES\x03"q\xf3\xbc\xe6\xc9\xf9S\xca\x98\xa6\x1d\xf7#\xe9\xb3\'\xc8A\xa4\x96\xce\xac\xe2V\xa4\x9d&R\x8bq\x12\x9f\xdd\x06\n\xc6\x13Q\xfd\xf6>\xab\xbag\xa2\x19VEc\xa2\x99\x0b\xe5\xdf\xd8\xbd\x80\x80\x83\x93\xa6\xc8\xc3\xc2\xa9\xf3hVfb\x00\xa6k\xc2g\xbf\xfe~\x12\xf5\x1c\xca\xdc\x11\n?\xae|@V\xb4\x02\xb2\x15\xcd\xa9h"\xb4\xc82\xde\xce\xe9\xddU\xc48\x96~\xf2\xdc\xa5[*\xf6\xba\xcf\xa2\x8d\x0cBmUb!\x91S\x15\x97\x8c\xcd\x8f\xe0\x87\xe5\x18\xaf\x07\xd6\x848HC\xac\xd0\xdd\x83\x87\xaf\xe1\xffB\xcfp\xbeP\xa8\xa5=J8U\x7fq\x8eNR\xda\x0c-\xc6;\xd6M\x9f\x0f\x88LV\xc7\x11eQ\xa9\x08Gv\xd7\xc7\x92\xa6$\x06\x08L\xb6\xdd\xd6X\x07@\xe8\xfa\xad\xdcZS\xe3\xf7\x17\xa3\xad\xe3\xb7b\xef\xfc\xcfI\x01\x9f\x9c\x01\x85\n^R\xbc\x84\xe84\xa7*\xf9\x96\x86}qw\xe8\x08\xa2E\x8fPZ\x88\x9d\x19\x02%$\xacQ`3\xaf\x92\x93\x12\x9c\x1e`\xd2}\xb6{X\xfe\xab5\x84\xc4\xafO\xe4\x96\x84O3\x17\x8a\xc1\xdc\x92*R\x89@\xc6\x85\xdf\xfe\xf9B\x80\x8a\x02Ou\x8f\xa8b\x9f\x1e*\x18\xde2M\xedI')
