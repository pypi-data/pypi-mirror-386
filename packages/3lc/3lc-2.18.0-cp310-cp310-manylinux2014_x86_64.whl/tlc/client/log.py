# Pyarmor 9.1.9 (pro), 005329, 3lc, 2025-10-22T14:52:11.743763
from tlcsaas.pyarmor_runtime_005329 import __pyarmor__
__pyarmor__(__name__, __file__, b'PY005329\x00\x03\n\x00o\r\r\n\x80\x00\x01\x00\x08\x00\x00\x00\x04\x00\x00\x00@\x00\x00\x00\xac\x07\x00\x00\x12\t\x06\x00\xb5\xf9\xf3\xd5\xdcC\xe7e8+Z\xca\xa7\\\xf3\xb4\x00\x00\x00\x00\x00\x00\x00\x00\xfew\xc1v\xea.\x16\\\xab\x88\xc9`\xa8%}\x0c\xf0i2\xca\x01B3\r!\xdb\xb0\x05~N\x87Y\xec\xbd\xe2}\xa1)\xbc\xac\xd1;\xcf\x9a\xbcBS\x9d\xe1J\xee\x9a\xb1\xcd\x97\xf3X\xd8l,\xb0\x8c\x15\xc1\xbbJ\xf7\xa8C\x7f\x90zv\xb8d\xd1\x82\xddlX\x9c\xc6\x96\xb2\x1ab\x87\x975X!\xa8Gt^\xa7?\xdb\x85\xbf\x9b\xf4\x84b\xa8:N\x1a\x9bH\x0c\r\x91/\xc2\xb8\xde;\xa6\xb1\xa2\xdd>\xba]\x83\x0e\xb3\x18\x90\xe8uyR\x80\xee\x80\x99%m\xba\x83\xbcb\xd6\xb1\x83.P\xdaB\xd6\';\xcb\xea\xc4M\xda7\xab\x05\xbf\x84\x84\x18x\x1a\x19\xac\xdc\xe9h\xfc[g\x94\xb7\x9bU\xc6\x03\xeb\x1a\xab\xdf1T\x884\xcc\xdbC\x93\x9fZM\x9cn\xc5\x9c\xc8\xfa\x15\x8d\x04\xa1\x84\xfc\xacw#H\x85\xf2Q\x03~=M\xba\xb6\x06M\xb3\x98\xe1\xf1\x8f4\xfc\xa7r\xc8\x92\x96\x8f\x9by\x15\xfa\xfed:\xa2\x8f\xa0\\\x0bpV\xc9\x1a\x00\xc9\xb4l\xfd\xd6\x89\xa6>\xef\xfe\xff\xc0\xd1\xbc\xc0\xc6\xa2\x91\x83O!F\xbfz\xc9\x17\x8f\xc27B>\x1a^\x89k\x80}(\xdc\xc29\x99\xe7\xc3\x95+@\xe1D\x84\xf9\x97\x08\xb1S\x0c\x0b\x15p\xb3\xf0\x83\xc7\x1f\xc0\xbc1\xa28p\xdc\x9e\xf5\x01\xd1\xdc\tdr/%\xc5\xd8r8\xc5\xf7\x9c\xe3E\xaeY\xee/\xbd\xd9\xe6\xcbj\xc4\xcdX\xcb~n8\x16\xdd\xa1\xdd\x97\x81\xd2\xad%0#\xdd0a\xc5\xb8:\xcf\xa2+Ex\x81\x92e\x08\x9f\x13:\xa2\xe9 [g\xae\x17\xaa\xcd9Ve\xf1V\x17eQ\xe2\xc3|\x86Mi\x81\xd7\xfb\t\xad\xa6\xae\xdc\x12vg\xc2h\xf7=\xc0\x192m\xb51\xdd>}<\xe8\x8e7\xa5\xa0\xf7S\xed\xc3L\x9c\xb7\xd8\xd2\x9cO\xfd\xe8s\xe3.\rz\x038\xccj62\xb3\xecR\x83\xfe$\x1f\xa4e\\\x00hl\x08\x9d\xc4wF\xdb]\xcc\xde\x99\xa9\x00]\xd3\x94ST\x15$\xda\x9c\xc0*#\x8f/\xb2^\xf1\xe8\xa3\x1e?\x82\xca\xe1\x80\xc1\xe0\xf1\x00\xce\xbf\xfd\xeb\xf5\xb28\xa0\xdcd\x90\xac\x8ckO\x80\xf6\x19\xc9#\x94\xadf\x0b\xbc~\x06G\xc3\x9b\xd9\x03\xab#\x04\xfddMM\x1ex\x04\xb1WQ\xe8\xe0\xcb8\xe3\xb0V\xff\xea\xbfd\xbbG^\x8cQ\x02\xb7\xd6b\xce^{\x16\x84\x0e\xd60\xd6\xc6\x10\r\xb4\xba\x83\xa0-w\x1e\xc3\x12\xcb\xdb\x19_\xab\xcb\x14\xfc\xcc\xa7\x84\x13\x067U\xf7u{\x06;\xa8\xa3g0\xb0,\xd6e\xcb\x00\xa1\x12\xcbZB\xce\xc8\xe1.\x82\xa5tEoz\t\xe4\x95\x81By\xe0Uc\xe6|\x08c\xd2\x80\xf8\xc0\x80\xf7HA\xc1\xc4\xbf\x1fG\x0e\xb6\x1eM\xac\'\xa2\xd9s\xfe\x17\x9b\x19F5r3\x80\x89\xf0\x90|\x99\x030\x12\xc0ea^M;qy\xc5\xd0M\x8c|vD\x0f\x8d!\xb8m\xb3a\x88\x02\x1f\xb1\x15n\x13\xcd\xcd\xac\x11\xdf\x7f4\r3\x89m\x8e\xdev?H\\\x82\x1a\x81\xa1\x11\xc9\x83\xf1\xe4\x9b\xcb6\x95\xb6\x16I\xb9\xd3g\x80\xe1\xc4\xa4\xc6\xf8\x8f\xb9s\xb9\x0e\xaa\x8f}\x1f=Y\xac\xb8Yl\x1f\x02|\xbd"\x89\xd2g}c\xb0K\xc9\x1e\xf3\x1d\xff>\xf0f>\xf2)\xaa\xf7%3%%F0`\xbc\x1es\xd2\xbb\x9f\x1e\xdb\x94\xed\xfb\xdf\xa7\xf1\x8d\x05\x00\xef\xdcF\xe9\x1c\x83\x93\xfc#\xd9*}\x0f\xd1\xa0A\xd02\xb4\x12\xb0\xff e\x83M\x869\x83\xa6\xbd/a\x12\xde\xbcj.9\xf9\xa5m\xaf/\xdf;\xb1\'\xdd\xf2%\xec\xc3\xceA\xd5\xc8\x17\xe9J\x99s\xb0O\xc4R\xf4|\x9fxX\x13g#T\xfff\xb8\x1e\x13\xb4\xc7\x89\x870 \xc5\xf6o\xfd\xf3\x7f%\xed\\\xad\xf4\x85\xfc=!\xc0>k\x1a\xd6\x90hx#w\x05]\xa5}kw\xdf \xd6\xaf\x89\xf4\x9f7\xd6Z\xcd\x9a\x10D\xb9\xef\x01(\x135"G$\x7f;\x98\xf3\xdf\x9b@T\xe4E3\t\xc5\\\xf1\x88\xb8Q\x892\xa0\xd3K\xc6\xad\x13O:\'\xd1\x06\x87X\xa9\xad\xe6\x18\x98p\x1f\xc3\x85eu\x0eL|\xfa\xbe\xfc\\\xfd\x82=\xbcC\x84xd\x0e\x89g^\xdd\xbf\xb2\xc9BI\xef\x15\xb2\xff\x92\xbe\xa5\xeb`@g\xab\xdf+\xce\xdf\xafx\x85\x88\xecc\x96\x9aR\x1a\xb0\xd5k\x8b\xf3\xc2\xd0\x80!\xe2\xa7Jp\x0c9D\x8f\x1dl\x99\x15\xfc\x82Q~\x9a\xf5WI\xf4\x046\xea\x1b\xce\xc5\xd6\xa2\xb2^\nc\xcet\xfa\x87\xbb\xe9\xf1J*\xf8]\xcd\xfc\r\xfc\xde]s\xfd\x005\x95%:\x10\xe5e\xd4i\xbf\x99#\x92\x8c0\x96@VO\xd8[\r\xc7\xb4?\xee\xb3PK\x0b)=/\xea\xe9;\x1c\x8e\x05jr\x9a\xb4U4\xb0k\x9d\x9dmV#\xca\x1cS\xbaO\xf2(\x80\x9f\xdbCK=T\xc2\x99\xe6\x90:\xe0s\xa3-\xda\x0b\x92Z,\xc6\x8f\x04\xde[m\xe2\x99\x87\xf8\xb9\xc4\xfbSl\xb6\xbc\xe6\t9Y\r\xb6{91-ce"\x9a\x15X\xdeZ\xcbs=m\x80B{\xeek`\n7:\xb6\x03YF\xba9\xbfE\xe4\xb4x\x11%G\x1c\xac\xe9\x86\x88\xd6\xe0E\xf7\x1aEF\xf5\x9a\x80\x95\xb7\xccK\xa4\xee\xc2\xd9\xa5\xdd]%\xb8+\xb4\xa0Y@L\x9fz\x80:y\x80p\x12\x0c\xa4\xbe\xe4\xcdO^\xc5&_\xb5\xc7&\r\x84\xe41\n\xc6g\xa983S\xf2\x10\xb2\xf4\xf9\xfeF\x13Y\x0b\xd9C9\x85\xeeu\x84\x02\xc6\xc20.cB\x15\xf9\x94\xe7\xaa \xd0\x0b\x11%jH\xe2\xd8\x14\x7f\xb6\x02\xfc\xdd\xa8\xc4\x05Z^\x8a\xf9P\x11\xdbl:\xd9\x8a\xeaw\xbb\xa3\xe6\xb70\x02\x00\xb0\xb95\xa6\x8b\xe8>\n.O\n\x156\x86g\xa6@w>2\xad\x02V\xac\x99zB\x15v]!\xc5\xf80\xabm\xdbn\x0b\xe0\xa6\x1c\xba\xd6\x86\x1c\xe7\x9f\xe0\x85T\xafHC\xac\x8f\xa4\x8e\xec\xa47#q\xadwsT\x0f<7\xbe<b3\x0f\x9f\xa1.=8\x9b\x13\xd5nH\xb2`\xbf\xc8\x92T\x88\xa7\x93\x85\xa4T\xef\xd5\xd7>\xdf<\xaaw"\xc1\xfc/\x00\xc4\x8a\x0e%\xf5\x81\xad\x88;\x88\xcc\x12\xae\xf3m\\Z\xa5-\x95\xff\x89a\xe1)\xbb\xe6\x87\xee\xabpW3\x8e\x85]\x05\x97%\xb3(\xf7\x12\r\x1e\xd2(\\e)\xec\n\xbeQ\x02Q\'\xaf\x12\xee\xf1\x1d\xfe\x9b\xfd\xb0\xa0\xf3:\xc3\x88\xc9\xe0\x80\xa5\xa4Zd\xbd\x89\x9bT\xd0JLc\xb1\xb2\xeb\xe9\x8c\xb3a7F\xd3\x90\xa9\xba5\xa9\xf9\xb4\xdeT\x0c\xd9n5\xc1\xf3\xa7\xc8\xe0n\x9d\xf8\xef9rf\xbe\x84\xbc\xa4\xb6m\xca\xf0\xb5\xe5\xec%\x9d\xa9\x07C#H\xc3\x82\xe3YM0dk\x90\xcb\xc6a#\xacr\xa8`\x8e\xc6\x93:,?>~\xf6Dm\n\xa1\x05\xfa\xae\xc0\x01\xf1\xee\xd8\xea\xd1!\x01\xb7\xb9)\x12\xc4\x84\x08\r\x91\xe0\xfdIdz>\xd3\xd6\x02\xc4\xdf\xd9g\x86d@\xa5rm)\x1d\x0f\xb4\x18Q\xe3\xc0t)\x17\xc0+n\xed\xa2\x94p\r\x16\xba\x08\x1c\x07\x18\x9a[\x89\xab\x9c\xf2\xa0\x07R\xb3\xe5\xeb?\xa8\x8f\x19yEb&\x10\xd9\xeb\xfa\xc8\x06s\x19\\\xaae\x83?\x05\xd6\x1f\x88\xa7,O1\x05^\xdf\x9e0R\x9c\xcf\x07}\xbd\x89\x83]\xae\xb9\xd1\x9c\xb7\x0f\x19\x0f\xd6\xbb\xd0\xf46t_\x12\xcd\xe2V\xb3<C\xb99@\xa4\x16\x1d\xc4\xa8i\x89[)\xc7@\x18\xbf#\x12v\x96\xc8cA\xd8\xa0\xb8\xa4\x1a\nD\xf3\xd3\xb4R0\xca\x7f\xa4\xd1\xc0`]\x15V\x03\x81\xaee\xe9\xfa\xf3\xf5\x81\xbd\x96\x91\xa1:$\x0c\xae\x81\xb8\xc2\xe7\xd9\xed\xc3P*\n#\xa1\xb0\xa8J\x06~\n\xbc\x89\x89\xae\x0f\x12\x8c\xa7\xfb8\xb2z\x8c\x81\xfe^\x8e\x83,-\xd5w\xef\xa8!\x0f\xa3\xb5\xfap\x9ct==!z\xdb\xb0\x80\x10"\xc3-\xaf$AQ\xf7\xd3C\x19X\x86\x13')
