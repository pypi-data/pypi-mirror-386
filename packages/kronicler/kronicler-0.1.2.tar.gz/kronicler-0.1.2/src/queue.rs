use super::capture::Capture;
use super::row::Epoch;
use log::info;
use pyo3::prelude::*;
use std::collections::VecDeque;
use std::sync::{Arc, RwLock};

#[pyclass]
pub struct KQueue {
    pub queue: Arc<RwLock<VecDeque<Capture>>>,
}

// Internal Rust methods
impl KQueue {}

#[pymethods]
impl KQueue {
    pub fn capture(&self, name: String, args: Vec<PyObject>, start: Epoch, end: Epoch) {
        let c = Capture {
            name,
            args,
            start,
            end,
            delta: end - start,
        };

        info!("Added {:?} to log", &c);

        // Concurrently add the capture to the queue to be consumed later
        {
            let mut q = self.queue.write().unwrap();
            q.push_back(c);
        }
    }

    #[new]
    pub fn new() -> Self {
        KQueue {
            queue: Arc::new(RwLock::new(VecDeque::new())),
        }
    }

    pub fn empty(&self) -> bool {
        self.queue.read().unwrap().is_empty()
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    use std::time::{SystemTime, UNIX_EPOCH};

    #[test]
    fn add_to_lfq_test() {
        let lfq = KQueue::new();

        let t1 = SystemTime::now()
            .duration_since(UNIX_EPOCH)
            .expect("Not outatime.")
            .as_nanos();

        let t2 = t1 + 100;

        // Check that it's empty before we add
        assert!(lfq.empty());

        lfq.capture("foo".to_string(), vec![], t1, t2);

        // Check that it's not empty after we add
        assert!(!lfq.empty());
    }

    #[test]
    fn queue_starts_empty() {
        let queue = KQueue::new();
        assert!(queue.empty());

        let q = queue.queue.read().unwrap();
        assert_eq!(q.len(), 0);
    }

    #[test]
    fn multiple_captures_in_order() {
        let queue = KQueue::new();
        let t1 = 1000u128;
        let t2 = 2000u128;
        let t3 = 3000u128;

        queue.capture("first".to_string(), vec![], t1, t1 + 100);
        queue.capture("second".to_string(), vec![], t2, t2 + 200);
        queue.capture("third".to_string(), vec![], t3, t3 + 150);

        let q = queue.queue.read().unwrap();
        assert_eq!(q.len(), 3);

        // Verify captures are in order
        assert_eq!(q[0].name, "first");
        assert_eq!(q[0].delta, 100);
        assert_eq!(q[1].name, "second");
        assert_eq!(q[1].delta, 200);
        assert_eq!(q[2].name, "third");
        assert_eq!(q[2].delta, 150);
    }

    #[test]
    fn capture_calculates_delta_correctly() {
        let queue = KQueue::new();
        let start = 5000u128;
        let end = 7500u128;
        let expected_delta = 2500u128;

        queue.capture("test_function".to_string(), vec![], start, end);

        let q = queue.queue.read().unwrap();
        let capture = &q[0];

        assert_eq!(capture.start, start);
        assert_eq!(capture.end, end);
        assert_eq!(capture.delta, expected_delta);
    }

    #[test]
    fn concurrent_access_safety() {
        use std::thread;

        let queue = Arc::new(KQueue::new());
        let mut handles = vec![];

        // Spawn multiple threads adding captures concurrently
        for i in 0..10 {
            let q = Arc::clone(&queue);
            let handle = thread::spawn(move || {
                let start = (i * 1000) as u128;
                let end = start + 100;
                q.capture(format!("thread_{}", i), vec![], start, end);
            });
            handles.push(handle);
        }

        // Wait for all threads to complete
        for handle in handles {
            handle.join().unwrap();
        }

        // Verify all captures were added
        let q = queue.queue.read().unwrap();
        assert_eq!(q.len(), 10);
        assert!(!queue.empty());
    }

    #[test]
    fn queue_preserves_capture_data() {
        let queue = KQueue::new();
        let name = "my_function".to_string();
        let start = 12345u128;
        let end = 67890u128;

        queue.capture(name.clone(), vec![], start, end);

        let q = queue.queue.read().unwrap();
        let capture = &q[0];

        assert_eq!(capture.name, name);
        assert_eq!(capture.start, start);
        assert_eq!(capture.end, end);
        assert_eq!(capture.args.len(), 0);
    }
}
