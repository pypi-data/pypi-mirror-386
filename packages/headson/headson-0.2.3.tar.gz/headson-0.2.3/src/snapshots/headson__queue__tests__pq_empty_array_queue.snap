---
source: src/queue.rs
assertion_line: 270
expression: "lines.join(\"\\n\")"
---
len=1
QueueItem { node_id: NodeId(0), parent_id: ParentId(None), kind: Array, depth: 0, index_in_array: None, key_in_object: None, priority: 1, number_value: None, bool_value: None, string_value: None } prio=1
